Code/Framework/AzFramework/AzFramework/Physics/RigidBodyBus.h

```diff
@@ -79,13 +81,66 @@ namespace Physics
 
     using RigidBodyRequestBus = AZ::EBus<RigidBodyRequests>;
 
-
+    //! Notifications interface for a rigid body (static or dynamic).
     class RigidBodyNotifications
         : public AZ::ComponentBus
     {
+    private:
+        template<class Bus>
+        struct RigidBodyNotificationsConnectionPolicy : public AZ::EBusConnectionPolicy<Bus>
+        {
+            static void Connect(
+                typename Bus::BusPtr& busPtr,
+                typename Bus::Context& context,
+                typename Bus::HandlerNode& handler,
+                typename Bus::Context::ConnectLockGuard& connectLock,
+                const typename Bus::BusIdType& id = 0)
+            {
+                AZ::EBusConnectionPolicy<Bus>::Connect(busPtr, context, handler, connectLock, id);
+
+                AZ::Entity* entity = nullptr;
+                AZ::ComponentApplicationBus::BroadcastResult(entity, &AZ::ComponentApplicationBus::Events::FindEntity, id);
+                if (entity)
+                {
+                    // Only immediately dispatch if the entity is already active, otherwise when
+                    // entity will get activated it will make the notifications itself.
+                    const AZ::Entity::State entityState = entity->GetState();
+                    if (entityState == AZ::Entity::State::Active)
+                    {
+                        // Only immediately dispatch if the entity is a RigidBodyRequestBus' handler.
+                        RigidBodyRequestBus::EnumerateHandlersId(
+                            id,
+                            [handler, id](const RigidBodyRequests* rigidBodyhandler)
+                            {
+                                if (rigidBodyhandler->IsPhysicsEnabled())
+                                {
+                                    handler->OnPhysicsEnabled(id);
+                                }
+                                else
+                                {
+                                    handler->OnPhysicsDisabled(id);
+                                }
+                                return true;
+                            });
+                    }
+                }
+            }
+        };
+
     public:
-        virtual void OnPhysicsEnabled() = 0;
-        virtual void OnPhysicsDisabled() = 0;
+        //! With this connection policy, RigidBodyNotifications::OnPhysicsEnabled and
+        //! RigidBodyNotifications::OnPhysicsDisabled events will be immediately
+        //! dispatched when a handler connects to the bus.
+        template<class Bus>
+        using ConnectionPolicy = RigidBodyNotificationsConnectionPolicy<Bus>;
+
+        virtual void OnPhysicsEnabled([[maybe_unused]] const AZ::EntityId& entityId)
+        {
+        }
+
+        virtual void OnPhysicsDisabled([[maybe_unused]] const AZ::EntityId& entityId)
```

Comment:
nit: As `AZ::EntityId` is just a wrapper around a `u64` it's generally preferred to pass by value (so it can be passed in a register)

Reply:
I'm always 50/50 on that one because internal knowledge of `EntityId` is needed to make that judgment. It's extremely unlikely, but using reference is a bit safer long term in case the internals of `EntityId` can change in the future. Since the comment starts with "nit" I'd leave it as it is for now.